api4/cloud.go

```diff
@@ -303,14 +302,14 @@ func getSubscriptionInvoicePDF(c *Context, w http.ResponseWriter, r *http.Reques
 		return
 	}
 
-	pdfData, appErr := c.App.Cloud().GetInvoicePDF(c.Params.InvoiceId)
+	pdfData, filename, appErr := c.App.Cloud().GetInvoicePDF(c.Params.InvoiceId)
 	if appErr != nil {
 		c.Err = model.NewAppError("Api4.getSuscriptionInvoicePDF", "api.cloud.request_error", nil, appErr.Error(), http.StatusInternalServerError)
 		return
 	}
 
 	err := writeFileResponse(
-		fmt.Sprintf("%s.pdf", c.Params.InvoiceId),
+		filename,
```

Comment:
Given that this could be empy, we should check for it and generate a filename if is not the case

Reply:
Added check in enterprise